src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityRockBreaker.java

```diff
@@ -34,28 +39,58 @@ protected RecipeLogicEnergy createWorkable(RecipeMap<?> recipeMap) {
         return result;
     }
 
-    protected boolean checkSidesValid(BlockStaticLiquid liquid) {
-        EnumFacing frontFacing = getFrontFacing();
+    @Override
+    public void onNeighborChanged() {
+        super.onNeighborChanged();
+        if (hasValidFluids)
+            return;
+
+        boolean hasLava = false;
+        boolean hasWater = false;
         for (EnumFacing side : EnumFacing.VALUES) {
+            if (hasLava && hasWater)
+                break;
+
             if (side == frontFacing || side == EnumFacing.DOWN || side == EnumFacing.UP)
```

Comment:
Instead of checking both `DOWN` and `UP`,  call `side.getAxis().isVertical()` and check.